Python/gc_free_threading.c

```diff
@@ -1704,8 +1709,13 @@ PyObject_GC_Del(void *op)
     }
 
     record_deallocation(_PyThreadState_GET());
-
-    PyObject_Free(((char *)op)-presize);
+    PyObject *self = (PyObject *)op;
+    if (_PyObject_GC_IS_SHARED(self)) {
```

Comment:
I think this condition is too general, but it might be fine for now. For example, this check will include shared dict and list objects, which don't generally have inline values.

I think we only need to delay freeing the object if:
1) It's marked as shared
2) It's an object with `Py_TPFLAGS_MANAGED_DICT` set on it's type (i.e., can have inline values)
3) Has a `__dict__` that outlives it

We might want to use another bit `ob_gc_bits` to track this

Reply:
Yep, I was thinking the same thing, particularly with having a different bit which I mentioned in the description.  I'll just go ahead and add a new bit now.